### ISSUE TITLE
Routing after user selection fails

### ISSUE THREAD

**Issue by NathanaelMangold** [TaleTime/TaleTime#148]
After relogging into an existing profile, it does not route to the main page

"select-user-profile.page.html" Line 12
```<ion-content class="ion-padding">
  <ion-list>
    <ion-item
        (click)="select($event, userProfile.id)"
        *ngFor="let userProfile of this.profilService.getUserProfiles()"
    >
      <ion-avatar slot="start">
        <img src="assets/imgs/profile/{{userProfile.avatar.name}}" />
      </ion-avatar>
      <h2>{{userProfile.name}}</h2>

      <ion-button *ngIf="isShowingOptions" (click)="delete($event, userProfile.id)" fill="clear" slot="end">
        {{ 'COMMON_DELETE' | translate }}
      </ion-button>
    </ion-item>
```

This call seems to be failing
"select-user-profile.page.ts" Line 43

```
public select(event, userProfileId: string) {
    event.stopPropagation();
    this.profilService.setActiveUserProfile(userProfileId).subscribe(
      (success) => {
        if (success) {
          // this.navCtrl.push(TabsPage);
          this.settingsService.loadSettings();
          this.router.navigate(["/tabs/story-menu"]);
        }
```

**Comment by TimReck** [comment]
this.profilService.setActiveUserProfile(userProfileId).subscribe is successful, but this error follows:

TypeError: this.activeUserProfile.getArrayOfStoriesByLanguage is not a function

story-menu.page.ts line 71
```
  ngOnInit() {
    this.stories = []
    this.activeUserProfile = this.profileService.getActiveUserProfile()
    if (this.activeUserProfile) {
      this.activeUserProfileName = this.activeUserProfile.name;
      this.activeUserProfileAvatarName = this.activeUserProfile.avatar.name;
      //todo Ändern, dass nur noch Enum anstelle von Strings benutzt werden
      if (this.languageService.selected == "de-DE") {
        this.stories = this.activeUserProfile.getArrayOfStoriesByLanguage(AvailableLanguage.German);
      } else {
        this.stories = this.activeUserProfile.getArrayOfStoriesByLanguage(AvailableLanguage.English);
      }
    }
    console.log(this.stories);
  }
```

The function getArrayOfStoriesByLanguage exists in userProfile.ts line 66
```
/**
   * Returns an array of type Array<StoryInformation> for a given language
   * @param {AvailableLanguage} lang Language as enum
   * @returns {Array<StoryInformation>}
   */
  public getArrayOfStoriesByLanguage(lang: AvailableLanguage): Array<StoryInformation> {
    return this.arrayOfStories.filter(o => o.language === lang)

  }
```

**Comment by TimReck** [comment]
story-menu.page.ts line 64
```
this.activeUserProfile = this.profileService.getActiveUserProfile();
+const userProfile = new UserProfile(this.activeUserProfile.name, this.activeUserProfile.avatar, this.activeUserProfile.child)
```

"as userProfile only tells the compiler that it's safe to assume that the value is of type userProfile, but doesn't have any effect on runtime and it won't have any methods."
https://stackoverflow.com/questions/42899570/method-in-typescript-class-give-error-is-not-a-function

fixes the error message, but creates new errors.

**Comment by SilasScholl** [comment]
Fixed in abf7f6f1bd1355ba8dd4b79323c3f1af708d8187
